src/storage/write_ahead_log/log_serializer_task.cpp

```diff
@@ -112,12 +112,15 @@ void LogSerializerTask::SerializeRecord(const terrier::storage::LogRecord &recor
       // ProjectedRowInitializer from these ids and their corresponding block layout.
       WriteValue(delta->NumColumns());
       WriteValue(delta->ColumnIds(), static_cast<uint32_t>(sizeof(col_id_t)) * delta->NumColumns());
-
+      // Write out the attr sizes. We write these out so we don't need the block layout during deserialization
```

Comment:
Do we need to do this?  We should have all of the information available on the other end.

Reply:
But you're replaying them in order, so you've replayed all of the catalog actions and can get the schema object by the time you might have inserts into a new table.  Right?

Reply:
Yes, it would require a sizable change to the serialization/deserialization. I mentioned in another comment I'd prefer to do this in another PR.